File: app/src/main/java/com/cjstudio/sosestrada/Prestador.java
package com.cjstudio.sosestrada;

public class Prestador {
    private String uid, nome, cnpj, telefone, email, servico, preco, logo;
    // Campos antigos (fallback)
    private String localizacao, endereco;
    // Novos campos
    private String rua, numero, bairro, cidade, complemento, estado, pais;
    private double distancia;
    // Status da solicitação (pendente, aceito, recusado ou null)
    private String statusSolicitacao;
    // Id da solicitação ativa com este prestador (usado para abrir o chat)
    private String solicitacaoId;
    // Mensagens do prestador ainda não lidas pelo motorista, na solicitação ativa acima.
    private int naoLidasMotorista;

    public Prestador() {}

    // Getters e Setters existentes...
    public String getUid() { return uid; }
    public void setUid(String uid) { this.uid = uid; }
    public String getNome() { return nome; }
    public void setNome(String nome) { this.nome = nome; }
    public String getCnpj() { return cnpj; }
    public void setCnpj(String cnpj) { this.cnpj = cnpj; }
    public String getTelefone() { return telefone; }
    public void setTelefone(String telefone) { this.telefone = telefone; }
    public String getEmail() { return email; }
    public void setEmail(String email) { this.email = email; }
    public String getServico() { return servico; }
    public void setServico(String servico) { this.servico = servico; }
    public String getPreco() { return preco; }
    public void setPreco(String preco) { this.preco = preco; }
    public String getLogo() { return logo; }
    public void setLogo(String logo) { this.logo = logo; }

    // Fallback getters/setters
    public String getLocalizacao() { return localizacao; }
    public void setLocalizacao(String localizacao) { this.localizacao = localizacao; }
    public String getEndereco() { return endereco; }
    public void setEndereco(String endereco) { this.endereco = endereco; }

    // Novos getters e setters
    public String getRua() { return rua; }
    public void setRua(String rua) { this.rua = rua; }
    public String getNumero() { return numero; }
    public void setNumero(String numero) { this.numero = numero; }
    public String getBairro() { return bairro; }
    public void setBairro(String bairro) { this.bairro = bairro; }
    public String getCidade() { return cidade; }
    public void setCidade(String cidade) { this.cidade = cidade; }
    public String getComplemento() { return complemento; }
    public void setComplemento(String complemento) { this.complemento = complemento; }
    public String getEstado() { return estado; }
    public void setEstado(String estado) { this.estado = estado; }
    public String getPais() { return pais; }
    public void setPais(String pais) { this.pais = pais; }

    public double getDistancia() { return distancia; }
    public void setDistancia(double distancia) { this.distancia = distancia; }

    // Getter e Setter para statusSolicitacao
    public String getStatusSolicitacao() { return statusSolicitacao; }
    public void setStatusSolicitacao(String statusSolicitacao) { this.statusSolicitacao = statusSolicitacao; }

    public String getSolicitacaoId() { return solicitacaoId; }
    public void setSolicitacaoId(String solicitacaoId) { this.solicitacaoId = solicitacaoId; }

    public int getNaoLidasMotorista() { return naoLidasMotorista; }
    public void setNaoLidasMotorista(int naoLidasMotorista) { this.naoLidasMotorista = naoLidasMotorista; }

    // Método que monta o endereço completo (com fallback)
    public String getEnderecoCompleto() {
        // Tenta usar os campos novos
        StringBuilder sb = new StringBuilder();
        if (rua != null && !rua.isEmpty()) sb.append(rua);
        if (numero != null && !numero.isEmpty()) sb.append(", ").append(numero);
        if (bairro != null && !bairro.isEmpty()) sb.append(" - ").append(bairro);
        if (cidade != null && !cidade.isEmpty()) sb.append(", ").append(cidade);
        if (estado != null && !estado.isEmpty()) sb.append(" - ").append(estado);
        if (pais != null && !pais.isEmpty()) sb.append(", ").append(pais);
        if (complemento != null && !complemento.isEmpty()) sb.append(" (").append(complemento).append(")");

        // Se não tem campos novos, usa os antigos
        if (sb.length() == 0) {
            if (endereco != null && !endereco.isEmpty()) {
                return endereco;
            }
            if (localizacao != null && !localizacao.isEmpty()) {
                return localizacao;
            }
            return "Endereço não informado";
        }
        return sb.toString();
    }
}

File: app/src/main/java/com/cjstudio/sosestrada/Prestador.kt
package com.cjstudio.sosestrada

import com.google.firebase.firestore.Exclude

// Documento prestadores/{uid}. Os campos do fim (distancia em diante) não
// vêm do Firestore: são preenchidos na busca do motorista (SocorroActivity).
data class Prestador(
    var uid: String? = null,
    var nome: String? = null,
    var cnpj: String? = null,
    var telefone: String? = null,
    var email: String? = null,
    var servico: String? = null,
    var preco: String? = null,
    var logo: String? = null,
    // Campos antigos de endereço (cadastros anteriores aos campos separados)
    var localizacao: String? = null,
    var endereco: String? = null,
    var rua: String? = null,
    var numero: String? = null,
    var bairro: String? = null,
    var cidade: String? = null,
    var complemento: String? = null,
    var estado: String? = null,
    var pais: String? = null,
    var distancia: Double = 0.0,
    // Status da solicitação com este prestador (pendente, aceito, recusado ou null)
    var statusSolicitacao: String? = null,
    // Id da solicitação ativa com este prestador (usado para abrir o chat)
    var solicitacaoId: String? = null,
    // Mensagens do prestador ainda não lidas pelo motorista, na solicitação ativa.
    var naoLidasMotorista: Int = 0
) {
    // Endereço montado dos campos novos; sem eles, cai nos campos antigos.
    @get:Exclude
    val enderecoCompleto: String
        get() {
            val partes = StringBuilder()
            if (!rua.isNullOrEmpty()) partes.append(rua)
            if (!numero.isNullOrEmpty()) partes.append(", ").append(numero)
            if (!bairro.isNullOrEmpty()) partes.append(" - ").append(bairro)
            if (!cidade.isNullOrEmpty()) partes.append(", ").append(cidade)
            if (!estado.isNullOrEmpty()) partes.append(" - ").append(estado)
            if (!pais.isNullOrEmpty()) partes.append(", ").append(pais)
            if (!complemento.isNullOrEmpty()) partes.append(" (").append(complemento).append(")")
            if (partes.isNotEmpty()) return partes.toString()
            return endereco?.takeIf { it.isNotEmpty() }
                ?: localizacao?.takeIf { it.isNotEmpty() }
                ?: "Endereço não informado"
        }
}
